src/api/Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM node:18.16.0
+
+# Create api directory
+WORKDIR /api
+
+# Install app dependencies
+# A wildcard is used to ensure both package.json AND package-lock.json are copied
+# where available (npm@5+)
+COPY package*.json ./
```

Comment:
package-lock.json is auto-generated based on your package.json so it shouldnt be copied into the container. You can add it to the dockerignore file and only `COPY . .` (one less copy step, no need to copy package separately).

Then once you start with multi-stage and targeted builds I can show you how using `ENV NODE_ENV=production` will tell the `npm install` command to omit dev dependencies.

Reply:
You would also want to add package-lock.json to the gitignore. You may also have to `git rm package-lock.json` so github stops keeping track of it.

It's also important that you don't introduce any more of those caret symbols (^) into the package.json and only specify exact version numbers for packages. Make sure everyone on your team has run the command `npm config set save-exact true`.

Reply:
All done and pushed.